samples/matter/common/src/bridge/ble_bridged_device.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "bridged_device_data_provider.h"
+
+#include <zephyr/bluetooth/addr.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+struct BLEBridgedDevice;
+
+class BLEBridgedDeviceProvider : public BridgedDeviceDataProvider {
+public:
+	explicit BLEBridgedDeviceProvider(UpdateAttributeCallback callback) : BridgedDeviceDataProvider(callback) {}
+	virtual ~BLEBridgedDeviceProvider() = default;
+
+	virtual bt_uuid *GetServiceUuid() = 0;
+	virtual void MatchBleDevice(BLEBridgedDevice *device) = 0;
+	virtual int ParseDiscoveredData(struct bt_gatt_dm *discoveredData) = 0;
+
+protected:
+	BLEBridgedDevice *mDevice;
+};
+
+struct BLEBridgedDevice {
+	using DeviceConnectedCallback = void (*)(BLEBridgedDevice *device, struct bt_gatt_dm *discoveredData,
+						 bool discoverySucceeded, void *context);
```

Comment:
Maybe we can use `BluetoothConnectionContext` if this is an only use case instead of `void`, we would spare additional casts in the callback.

Reply:
I don't think we should do that. First of all the `BluetoothConnectionContext` contains `shell *` and the `BLEBridgedDevice` doesn't depend on the application implementation. Generally the application may want to pass some custom content to the callback context, so I think keeping it void * is a proper design.